sendmail.py

```diff
@@ -0,0 +1,39 @@
+import email
+import sys
+import traceback
+import uuid
+from email.message import Message
+from pathlib import Path
+
+script_dir = Path(__file__).resolve().parent
+cwd = Path().resolve()
+
+
+def main():
+    if len(sys.argv) > 1:
+        mail_dir = sys.argv[1]
+    else:
+        mail_dir = script_dir / 'emails'
+
+    text = sys.stdin.read()
+
+    mail_dir = Path(mail_dir).resolve()
+    msg = email.message_from_string(text)  # type: Message
+
+    file = (mail_dir / '{}.mail'.format(uuid.uuid4()))
+    with file.open('wb') as f:
```

Comment:
What does this create the file as permissions wise? It wouldn't be a bad idea for this to be `-r--------` which is `400`

Reply:
The permissions are controlled by the umask on the process, and in this case will be owned by whatever user Anope is running as, but the daemon will be a different user.

Reply:
`660` could work and then I can run the daemon with the anope group

Reply:
You could run the daemon as setuid for anope's execution. And it could set umask when it starts, though open() in python lets you set the modes on the file anyway. Not sure if that switches umask back and fourth or just changes the modes afterwards. Having it group readable is still better than default, but having it entirely clean would be nice.